Tests/NIOHTTP2Tests/ConfiguringPipelineInlineMultiplexerTests.swift

```diff
@@ -71,7 +71,10 @@ class ConfiguringPipelineInlineMultiplexerTests: XCTestCase {
             )
         )
 
-        clientMultiplexer.createStreamChannel(promise: nil) { channel in
+        let errorHandler = ErrorEncounteredHandler()
+        let streamChannelPromise = self.clientChannel.eventLoop.makePromise(of: Channel.self)
+        clientMultiplexer.createStreamChannel(promise: streamChannelPromise) { channel in
+            try? channel.pipeline.addHandler(errorHandler).wait()
```

Comment:
This stream channel initializer will be executed on the event-loop; you can't `wait()` while on an EL. You need to use an alternative API, like the sync operations.

Reply:
> This one got marked as resolved but wasn't actually resolved.
Ah, sorry I fixed it in a bunch of places but missed a few others.

> FWIW I also don't think it's necessary for all of these tests to check this behaviour [..] I'd prefer we had a separate test which asserts the specific behaviour.
We were already performing assertions on the `close` promise everywhere - you could use the same argument there. I thought that it was more consistent if we are going to assert something on the `close` promise that we also check the right behaviour is seen for the `closeFuture` and the errors being thrown downstream. But I can revert those changes if you disagree and add a single test.

Reply:
No worries, I saw the first one and assumed they'd all been missed.

I'm okay not reverting it, the work's been done now. I get the following the existing pattern argument; I just think we should be mindful when we make changes like this to keep tests maintainable. This package is bad enough as it is as most tests have about three different versions because of the different multiplexing APIs...